enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,1029 @@
+subnet configuration.
+
+1. Cluster admin creates an install-config with `platform.aws.vpc.subnets` specified
+   with the subnet(s) while ensuring `platform.aws.vpc.subnets[].rolesOverride` is left empty:
+    ```yaml
+    apiVersion: v1
+    baseDomain: devcluster.openshift.com
+    metadata:
+      name: my-cluster
+    #...
+    platform:
+      aws:
+        region: us-east-1
+        vpc:
+          subnets:
+          - id: subnet-0fcf8e0392f0910d2 # Private / AZ us-east-1a
+          - id: subnet-0fcf8e0392f0910d3 # Public / AZ us-east-1a
+    #...
+    ```
+2. The OpenShift Installer installs the cluster into the VPC where the subnets exist and automatically determines
+   the role or purpose of each subnet. For IngressControllers, no specific subnet is explicitly assigned so that
+   the AWS CCM will use [AWS Subnet Discovery](#aws-subnet-discovery) when creating the load balancer for the
+   `default` IngressController.
+
+### API Extensions
+
+This proposal doesn't add any API extensions.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+This enhancement is directly enabling the https://issues.redhat.com/browse/XCMSTRAT-545
+feature for ROSA Hosted Control Planes.
+
+#### Standalone Clusters
+
+This proposal does not have any special implications for standalone clusters.
+
+#### Single-node Deployments or MicroShift
+
+This proposal does not have any special implications for single-node
+deployments or MicroShift.
+
+### Risks and Mitigations
+
+- **Risk**: Using public IngressControllerLB subnets for private clusters.
+  - **Mitigation**: We will reject private cluster installations using public IngressControllerLB subnets (see
+    [Consistent Cluster Scope with IngressControllerLB Subnets Validation](#consistent-cluster-scope-with-ingresscontrollerlb-subnets-validation))
+  - **Mitigation**: For private cluster installations using subnet discovery for IngressControllers, public subnets will be excluded from the 
+    `default` IngressControllers, provided the subnets are correctly tagged.
+- **Risk**: Using public ControlPlaneInternalLB subnets which would expose the internal control plane API load balancer.
+  - **Mitigation**: We will reject using public ControlPlaneInternalLB subnets (see
+    [Consistent Cluster Scope with ControlPlaneLB Subnets Validation](#consistent-cluster-scope-with-controlplanelb-subnets-validation)).
+
+### Drawbacks
+
+- Deprecating `platform.aws.subnets` will be painful for users.
+- Distinguishing between BYO subnets/VPC and installer-created subnets/VPC may be confusing if we need
+  to expand the `platform.aws.vpc.subnets` field in the future to accommodate installer-created
+  subnets configuration.
+  - For example, there may be a future desire to specify the IngressController role for installer-created subnets,
+    and it is unclear how this API design would accommodate that.
```

Comment:
This is a good point, but I **do think** the API design is extensible enough to handle this. An example implementation would be to make the subnet id optional, and when omitted the installer would create the subnet, and add an availability zone field. Something like:

```yaml
    platform:
      aws:
        region: us-east-1
        vpc:
          subnets:
          - availabilityZone: us-east-1a
            rolesOverride:
            - IngressControllerLB
            - ControlPlaneExternalLB
          - availabilityZone: us-east-1a
            rolesOverride:
            - ControlPlaneInternalLB
            - ClusterNode
```